lib/Doctrine/Migrations/DependencyFactory.php

```diff
@@ -235,10 +235,31 @@ public function getSchemaDumper() : SchemaDumper
         });
     }
 
-    private function getSchemaProvider() : SchemaProvider
+    public function hasSchemaProvider(): bool
+    {
+        if ($this->hasEntityManager()) {
+            return true;
+        }
+
+        if ((bool) $this->getConfiguration()->getSchemaProvider()) {
+            return true;
+        }
+
+        return false;
+    }
+
+    public function getSchemaProvider() : SchemaProvider
     {
         return $this->getDependency(SchemaProvider::class, function () : SchemaProvider {
-            return new OrmSchemaProvider($this->getEntityManager());
+            if ($this->hasEntityManager()) {
```

Comment:
Maybe use custom provided schema provider, when it is explicitly specified and fallback to check ORM otherwise?

WDYT?

Reply:
Implementations will most likely choose either ORM or a custom schema for DBAL. But I agree that the explicit configured one should have higher priority.